ext2/libext2.c

```diff
@@ -56,7 +57,9 @@ int libext2_handler(void *fdata, msg_t *msg)
 			break;
 		}
 
-		if (ext2_lookup(fs, msg->i.create.dir.id, msg->i.data, (uint8_t)strlen(msg->i.data), &msg->o.create.oid.id, &dev) > 0) {
+		/* FIXME: casting namelen to (uint8_t) is not a good way to check for length constraints */
+		namelen = strlen(msg->i.data);
+		if (ext2_lookup(fs, msg->i.create.dir.id, msg->i.data, namelen, &msg->o.create.oid.id, &dev) > 0) {
```

Comment:
If I had to nitpick - we don't check data we received from the outside - msg->i.data could be NULL. Same goes with namelen being too big - we could just give up and throw an error at the client.

Reply:
yeah, ext2 code is buggy in many places, we need serious unit tests for all available FS messages and test every filesystem with them.

I tried to add symlinks support to ext2 without minimal changes as it's the only option to test this code on ia32-generic. I have hoped for `dummyfs`, but it turns out it's broken: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/114